File: problems/93/solution.java

class Solution {
	public ArrayList<ArrayList<Integer>> generate(int a) {
		ArrayList<ArrayList<Integer>> result = new ArrayList<ArrayList<Integer>>();
		for (int i = 0; i < a; i++) {
			ArrayList<Integer> arrayList = new ArrayList<Integer>();
			arrayList.add(1);
			for (int r = 1; r <= i; r++) {
				int next = (arrayList.get(r - 1) * (i  + 1 - r)) / r;
				arrayList.add(next);
			}
			result.add(arrayList);
		}
		return result;
	}
}

File: problems/93/solution.cpp

vector<vector<int> > Solution::generate(int A) {
	vector<vector<int> > result(A);
	for (int i = 0; i < A; i++) {
		vector<int> v(i + 1); 
		v[0] = 1;
		for (int r = 1; r <= i; r++) {
			int next = (v[r - 1] * (i  + 1 - r)) / r;
			v[r] = next;
		}
		result[i] = v;
	}
	return result;
}

